src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java

```diff
@@ -44,9 +43,12 @@ public Jedis getConnection() {
         if (result.equalsIgnoreCase("pong")) return jedis;
 
         pool.returnBrokenResource(jedis);
-      } catch (JedisConnectionException ex) {
+      } catch (JedisException ex) {
         if (jedis != null) {
-          pool.returnBrokenResource(jedis);
+          try {
```

Comment:
Regarding connection is not marked broken: please note that if ping() does not throw, but returns anything else than "pong", the same method returnBrokenResource() is invoked.
And if ping() returns null, we get uncaught NPE there.
I suppose, both cases should not occur with sane redis server, but anything may happen in the wild :)